### PR TITLE
Upgraded angular router to use ui-router

### PR DESCRIPTION
**Pull request by paymog**
We now use ui-router instead of ngRoute. Karma tests
have also been updated to reflect these changes

I've also added a nested partial view in `users.jade` as an example.

### CODE REVIEW

**Comment by kjrcda** [comment]
that's dope. weird about what fixes that problem you were having.
Only thing I want to mention is this stuff doesn't work in IE, but it does in Firefox/chrome (aka doesn't really matter but just an FYI)

File: webapp/ng_tests/controllers/main_page_controller_test.js [hunk 1]
```diff
@@ -2,11 +2,12 @@
 
 describe('MainPageController', function () {
     beforeEach(module('lunchr'));
+    beforeEach(module('stateMock')); //mock the $state service of ui-router
 
     var DEFAULT_EMAIL = 'user';
     var DEFAULT_PASSWORD = 'password';
 
-    var $controller, $httpBackend, $rootScope, $location;
+    var $controller, $httpBackend, $rootScope, $state;
```
Comment (kyleleung75) [inline webapp/ng_tests/controllers/main_page_controller_test.js:10]: Why did you change from using location to state?

Comment (paymog) [inline webapp/ng_tests/controllers/main_page_controller_test.js:10]: Before, we would navigate from page to page using $location which would alter the url which angular would pick up.

Now, $state tells angular to go to a specific state which then gets reflected by a URL change.
